in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,326 @@
+
+  sign_parser = subparsers.add_parser('sign', help='Sign the link/layout '
+                                      'file.')
+
+  sign_parser.add_argument("signablepath", type=str,
+                            help="path to the signable file")
+
+  sign_parser.add_argument("-r", "--replace-sig", action="store_true",
+                            help="Replace all the old signatures, sign "
+                                 "with the given key, and add the new "
+                                 "signature in file")
+
+  sign_parser.add_argument("-i", "--infix", action="store_true",
+                            help="Infix keyID in the filename while "
+                                 "dumping, when -i the file will be dumped "
+                                 "as original.<keyID>.link/layout, "
+                                 "else original.link/layout")
+
+  sign_parser.add_argument("-v", "--verbose", dest="verbose",
+                            help="Verbose execution.", default=False,
+                            action="store_true")
+
+  sign_parser.add_argument("-d", "--destination", type=str, help="Filepath "
+                           "to which the output file should be dumped")
+
+  sign_parser.add_argument("-k", "--keys", nargs="+", type=str, required=True,
+                            help="Path to the key file/files")
+
+  verify_parser = subparsers.add_parser('verify', help='Verify the '
+                                        'link/layout file.')
+
+  verify_parser.add_argument("signablepath", type=str,
+                            help="path to the signable file")
+
+  verify_parser.add_argument("-v", "--verbose", dest="verbose",
+                            help="Verbose execution.", default=False,
+                            action="store_true")
+
+  verify_parser.add_argument("-k", "--keys", nargs="+", type=str,
+                            required=True, help="Path to the key file/files")
+
+
+  args = parser.parse_args()
+  return args
+
+def main():
+  """
+  First calls parse_args to parse the arguments, and then calls either
+  add_sign or add_replace depending upon the arguments. Based on the
+  arguments it then dumps the corresponding file.
+  """
+  args = parse_args()
+  length = len(args.keys)
+  rsa_key = in_toto.util.import_rsa_key_from_file(args.keys[length-1])
+  source_file_name = os.path.basename(args.signablepath)
+
+  if args.verbose:
+    log.logging.getLogger().setLevel(log.logging.INFO)
+
+  if args.subparser_name == 'sign':
+    try:
+      if args.destination and args.infix:
+        log.error('Please specify one and only one out of infix and '
+                  'destination!')
+        sys.exit(1)
+      else:
+        if args.replace_sig:
+          signable_object = replace_sign(args.signablepath, args.keys)
+        else:
+          signable_object = add_sign(args.signablepath, args.keys)
+
+        if signable_object._type == 'layout':
```

Comment:
`os.path.join` only really makes sense if you pass it two arguments, also you should only warn if `--infix` is specified and in general you could make this a little slicker. What do you think about the following?

```python
if args.infix: 
  log.warn('Ignoring option --infix for layouts...')

if args.destination:
  path = args.destination

else:
  path = args.signablepath

log.info("Dumping layout to '{}' ...".format(path))
signable_object.dump(path)
sys.exit(0)
```